src/chi/javascript/components/mobile-navigation.js

```diff
@@ -0,0 +1,458 @@
+    const self = this;
+
+    if (firstLevelDrawer) {
+      const firstLevelCloseButton = firstLevelDrawer.querySelector(`.${chi.classes.CLOSE}`);
+
+      this._firstLevelDrawer = Drawer.factory(this._elem, {
+        animated: true,
+        target: firstLevelDrawer
+      });
+      this._addEventHandler(
+        firstLevelCloseButton,
+        'click',
+        () => {
+          self._hideMobileNav();
+        }
+      );
+    }
+
+    Array.prototype.forEach.call(
+      firstLevelMenuItems,
+      (menuItem) => {
+        const drawerId = menuItem.getAttribute('href');
+        const drawer = document.querySelector(`.chi-drawer${drawerId}`);
+
+        if (drawer) {
+          const returnBackButton = drawer.querySelector(`.${DRAWER_HEADER_CLASS} button.${chi.classes.BACK}`);
+          const firstLevelContent = self._mobileNavElement.querySelector(`.${FIRST_LEVEL_DRAWER_CONTENT_CLASS}`);
+
+          this._addEventHandler(
+            menuItem,
+            'click',
+            function(e) {
+              e.preventDefault();
+              Util.addClass(drawer, chi.classes.ACTIVE);
+              Util.addClass(firstLevelContent, SECOND_LEVEL_SHOWN_CLASS);
+            });
+
+          this._addEventHandler(
+            returnBackButton,
+            'click',
+            function(e) {
+              e.stopPropagation();
+              Util.removeClass(drawer, chi.classes.ACTIVE);
+              self.resetActiveDrawerMenuItem();
+              Util.removeClass(firstLevelContent, SECOND_LEVEL_SHOWN_CLASS);
+            });
+          Util.addClass(menuItem, DRAWER_SUBITEM_LIST_TRIGGER_CLASS);
+          this._drawers.push(drawer);
+        } else {
+          this._addEventHandler(
+            menuItem,
+            'click',
+            () => self._activateFirstLevelMenuItem(menuItem)
+          );
+        }
+      }
+    );
+    Array.prototype.forEach.call(
+      firstLevelDropdownTriggers,
+      (dropdown) => {
+        const dropdownElemId = dropdown.getAttribute('data-target');
+
+        Dropdown.factory(dropdown, { dropdownElem: document.querySelector(`#${dropdownElemId}`) });
+      }
+    );
+    this._initSecondLevelDrawers();
+    this._addEventHandler(
+      this._firstLevelDrawer._drawerElem,
+      'chi.drawer.hide',
+      function() {
+        const firstLevelDrawerContent = self._mobileNavElement.querySelector(`.${FIRST_LEVEL_DRAWER_CONTENT_CLASS}`);
+        const activeSecondLevelDrawer = self._mobileNavElement.querySelector(`.${MOBILENAV_DRAWERS_CLASS} .${DRAWER_CLASS}.${chi.classes.ACTIVE}`);
+
+        if (activeSecondLevelDrawer) {
+          setTimeout(
+            function() {
+              Util.removeClass(activeSecondLevelDrawer, chi.classes.ACTIVE);
+              Util.removeClass(firstLevelDrawerContent, SECOND_LEVEL_SHOWN_CLASS);
+            },
+            500
```

Comment:
Please, use a constant. Why it is 500 and not other value. If it comes from another animation, that animation and this one should use the same constant.